tests_e2e/tests/ext_sequencing/ext_sequencing.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This test adds extensions with multiple dependencies to a VMSS using the 'provisionAfterExtensions' property and
+# validates they are enabled in order of dependencies.
+#
+import copy
+import uuid
+from datetime import datetime
+from typing import List, Dict, Any
+
+from assertpy import fail
+from azure.mgmt.compute.models import VirtualMachineScaleSetVMExtensionsSummary
+
+from tests_e2e.tests.ext_sequencing.ext_seq_test_cases import add_one_dependent_ext_without_settings, add_two_extensions_with_dependencies, \
+    remove_one_dependent_extension, remove_all_dependencies, add_one_dependent_extension, \
+    add_single_dependencies, remove_all_dependent_extensions, add_failing_dependent_extension_with_one_dependency, add_failing_dependent_extension_with_two_dependencies
+from tests_e2e.tests.lib.agent_test import AgentVmssTest, TestSkipped
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.resource_group_client import ResourceGroupClient
+from tests_e2e.tests.lib.ssh_client import SshClient
+
+
+class ExtSequencing(AgentVmssTest):
+    # Cases to test different dependency scenarios
+    test_cases = [
+        add_one_dependent_ext_without_settings,
+        add_two_extensions_with_dependencies,
+        remove_one_dependent_extension,
+        remove_all_dependencies,
+        add_one_dependent_extension,
+        add_single_dependencies,
+        remove_all_dependent_extensions,
+        add_failing_dependent_extension_with_one_dependency,
+        add_failing_dependent_extension_with_two_dependencies
+    ]
+
+    @staticmethod
+    def get_dependency_map(extensions: List[Dict[str, Any]]) -> Dict[str, Dict[str, Any]]:
+        dependency_map: Dict[str, Dict[str, Any]] = dict()
+
+        for ext in extensions:
+            ext_name = ext['name']
+            provisioned_after = ext['properties'].get('provisionAfterExtensions')
+            depends_on = provisioned_after if provisioned_after else []
+            # We know an extension should fail if a script was provided
```

Comment:
In our test cases, we only use "script" in the properties for extensions we want to fail. These scripts return a non-zero exit code. I'll update this since now I'm using commandToExecute to fail the extension

Reply:
but that is too "hacky"/limited, right? what if in the future we want a script that succeeds?

Reply:
i'll check specifically if commandToExecute contents are "exit 1"

Reply:
not perfect, but fair enough :)

please document that

Reply:
to clarify, the presence of 'exit 1' does not necessarily mean the script will fail (e.g. it may be within a conditional statement), but these scripts should be simple so this approach should work.

the keyword above being "should", hence the ask for documentation, since this may take somebody by surprise and the comment may reduce debugging time is something gets bitten